runtime/fiber.c

```diff
@@ -515,6 +478,23 @@ CAMLprim value caml_continuation_use (value cont)
   return v;
 }
 
+CAMLprim value caml_clone_continuation (value cont)
+{
+  caml_fatal_error ("caml_clone_continuation is unsupported");
+  return Val_unit;
+}
+
+CAMLprim value caml_update_handler (value stack, value hval, value hexn,
```

Comment:
This function worried me from a multi-thread safety angle; if two domains tried to update the handler at the same time then things would go bad.
 
It is actually in this PR because `caml_update_handler` is only ever called like this:
```ocaml
      let stack = take_cont_noexc k in
      let stack' = update_handler stack handler.retc handler.exnc effc in
```
Might it be better to have a `take_cont_and_update_handler` primitive that ensures that it wraps up the continuation take as well?
Alternatively maybe a comment to suggest that you should never call `caml_update_handler` without first ensuring that you own the stack.

Reply:
Combined them into a single primitive.